nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/work/TestWorkManagerImpl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2012-2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.ecm.core.work;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.work.api.Work;
+import org.nuxeo.ecm.core.work.api.WorkManager;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.FileEventsTrackingFeature;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeFeature.class, FileEventsTrackingFeature.class })
+@Deploy("org.nuxeo.ecm.core.event")
+@Deploy("org.nuxeo.ecm.core.event.test:test-workmanager-config.xml")
+public class TestWorkManagerImpl {
+
+    @Inject
+    public WorkManager service;
+
+    @Test
+    @Deploy("org.nuxeo.ecm.core.event.test:test-shutdown-delay.xml")
+    public void testShutdownDurationWithDelay() {
+        testShutdownDuration();
+    }
+
+    @Test
+    public void testShutdownDurationWithoutDelay() {
```

Comment:
With or without delay you test the same thing. Tests should highlight usage (by declaration) of `SHUTDOWN_DELAY_KEY`.

Reply:
I believe you missed something, the @deploy actually declares a value for the mentioned key through contribution to the ConfigurationService

Reply:
Yes, but in both cases behavior is the same. So we don't test that something change if we configure a delay.